### ISSUE TITLE
[REVIEW]: The SuiteSparse Matrix Collection Website Interface

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#1244]
**Submitting author:** @ScottKolo (<a href="http://orcid.org/0000-0001-8340-0824">Scott Kolodziej</a>)
**Repository:** <a href="https://github.com/ScottKolo/suitesparse-matrix-collection-website" target ="_blank">https://github.com/ScottKolo/suitesparse-matrix-collection-website</a>
**Version:** JOSS-1.0.2
**Editor:** @jedbrown
**Reviewer:** @dblarremore, @tkonolige
**Archive:** <a href="https://doi.org/10.6084/m9.figshare.7381076" target="_blank">10.6084/m9.figshare.7381076</a>

## Status

[![status](http://joss.theoj.org/papers/965b1c7bba801c85e9341a3db9013ef8/status.svg)](http://joss.theoj.org/papers/965b1c7bba801c85e9341a3db9013ef8)

Status badge code:

```
HTML: <a href="http://joss.theoj.org/papers/965b1c7bba801c85e9341a3db9013ef8"><img src="http://joss.theoj.org/papers/965b1c7bba801c85e9341a3db9013ef8/status.svg"></a>
Markdown: [![status](http://joss.theoj.org/papers/965b1c7bba801c85e9341a3db9013ef8/status.svg)](http://joss.theoj.org/papers/965b1c7bba801c85e9341a3db9013ef8)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/ScottKolo/suitesparse-matrix-collection-website" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@dblarremore & @tkonolige, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.theoj.org/about#reviewer_guidelines. Any questions/concerns please let @jedbrown know.

✨ **Please try and complete your review in the next two weeks** ✨ 

### Conflict of interest

- [x] As the reviewer I confirm that I have read the [JOSS conflict of interest policy](https://github.com/openjournals/joss/blob/master/COI.md) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/ScottKolo/suitesparse-matrix-collection-website">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Version:** JOSS-1.0.2
- [x] **Authorship:** Has the submitting author (@ScottKolo) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?

### Conflict of interest

- [x] As the reviewer I confirm that I have read the [JOSS conflict of interest policy](https://github.com/openjournals/joss/blob/master/COI.md) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/ScottKolo/suitesparse-matrix-collection-website">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Version:** JOSS-1.0.2
- [x] **Authorship:** Has the submitting author (@ScottKolo) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @dblarremore, it looks like you're currently assigned as the reviewer for this paper :tada:.

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01244/joss.01244/10.21105.joss.01244.pdf)

**Comment by jedbrown** [comment]
@dblarremore @tkonolige :wave: Welcome and thanks for agreeing to review! The comments from @whedon above outline the review process, which takes place in this thread (possibly with issues filed in the project repository). I'll be watching this thread if you have any questions.

**Comment by tkonolige** [comment]
@ScottKolo I think https://doi.org/10.1007/978-1-5041-2940-4_9 is the correct doi for Matrix market: A web resource for test matrix collections. It would be great if you can add that.

I also see that there is no documentation for building, installing, and testing locally. I created an issue for it in your repository: ScottKolo/suitesparse-matrix-collection-website#5.

**Comment by tkonolige** [comment]
@jedbrown I'm not sure about the example usage and functional documentation parts of the checklist. Does there need to be documentation on how to use the website? I think it is pretty straight forward.

**Comment by jedbrown** [comment]
I don't think human "use" of the website needs separate documentation, but there appears to be an API used by several external packages and I don't immediately notice documentation of that API.

**Comment by ScottKolo** [comment]
> I don't think human "use" of the website needs separate documentation, but there appears to be an API used by several external packages and I don't immediately notice documentation of that API.

We do include information about other interfaces to the Collection on the website itself on the [Interfaces page](https://sparse.tamu.edu/interfaces). I also just added information in the repository README referencing the Interfaces page and also the URL routing feature, which might be considered an API.

The use of most of the website should be (*we hope*) easy enough to understand and figure out. We also include tooltips to help users with certain terminology:

![image](https://user-images.githubusercontent.com/3859035/52680960-dbc26880-2eff-11e9-8d7f-f72e38e803dd.png)

Regarding the inner workings of things like UFget/ssget, we considered that somewhat out of scope for this paper as these interfaces were presented in [Davis and Hu](https://doi.org/10.1145/2049662.2049663) and are not strictly considered part of the web application. For example, these interfaces often provide the option to download a matrix, and that download URL is the same URL being used by the web application when you click the button on the website. The files are also not served by the Rails web application we've presented - they're being served by a file server. The way we see it, this is a "frontend" to access the "backend" presented in [Davis and Hu](https://doi.org/10.1145/2049662.2049663).

**Comment by tkonolige** [comment]
That all makes sense. I think you're good on the documentation front then.

@ScottKolo, @jedbrown should the release version listed here be bumped to match what is currently in the repository (now that some changes have been made)?

**Comment by jedbrown** [comment]
Yeah, we'll ask for a tagged release at acceptance, and the version will be set at that time.

**Comment by dblarremore** [comment]
**Trivial**: "applicaiton" typo in README. 

**Perhaps more useful**:   Possible change needed in installation instructions for OSX.
I tried to do the local installation, updated Ruby, and got all of that sorted out. On README line 
`bundle exec rake db:setup`
I kept getting an error
`PG::ConnectionBad: could not connect to server: No such file or directory`
`	Is the server running locally and accepting`
`	connections on Unix domain socket "/tmp/.s.PGSQL.5432"?`
I eventually fixed this by (apparently) manually starting the server via the top answer [here](https://dba.stackexchange.com/questions/75214/postgresql-not-running-on-mac) whose details are then described in detail [here](https://stackoverflow.com/questions/7975556/how-to-start-postgresql-server-on-mac-os-x). 
After that, my local installation was totally fine. 

To recap, my experience required that I insert this line between the bundle install and bundle exec rake commands:
`pg_ctl -D /usr/local/var/postgres -l /usr/local/var/postgres/server.log start`

Now, the question: is this worth inserting into the documentation? Are there files that could be modified by @ScottKolo or is this a sort of novice ruby user thing? (I am basically a test case for _user who does not know what he is doing with a local ruby installation_ so perhaps not an issue.) Curious to hear what you guys think.

**Comment by ScottKolo** [comment]
> To recap, my experience required that I insert this line between the bundle install and bundle exec rake commands:
`pg_ctl -D /usr/local/var/postgres -l /usr/local/var/postgres/server.log start`

After installing PostgreSQL, one needs to start it up so it's running in the background. You can do this a number of ways: manually start/stop, use the daemon, etc. After installing PostgreSQL from a package manager, it often spits out a line at the end about this (i.e. "run this magical incantation to start the daemon").

I'm happy to add whatever you think is helpful to the documentation. I might suggest the following additional call to `pg_isready`:

```shell
ruby -v                              # Check that Ruby 2.5+ is installed
pg_isready                           # Check that PostgreSQL database is ready
git clone https://github.com/ScottKolo/suitesparse-matrix-collection-website.git
cd suitesparse-matrix-collection-website
gem install bundler:1.16.6           # Dependency manager
bundle install --without production  # Download and install dependencies
bundle exec rake db:setup            # Generate and seed the database
bundle exec rails server             # Start the Ruby on Rails server application
```

Unfortunately, if `pg_isready` fails, it is not very informative regarding how to fix the problem.

I can add a line mentioning something about "PostgreSQL must be running - if it is not, try running `pg_ctl -D /usr/local/var/postgres -l /usr/local/var/postgres/server.log start`."

My only concern is that the above command assumes the user's `postgres` installation directory and log file location.

**Comment by dblarremore** [comment]
I like the idea to use `pg_isready` but how about putting this snippet above, under the bullet list that includes the need for ruby and postgreSQL?  Then it would be filed in with the prep work and dependencies instead of with the installation itself. 

What you wrote was really nice, so I tinkered with it a little, but maybe something like this? Happy for any modifications to this, since it's your README, but thought this might be useful:

> After installing PostgreSQL, note that one needs to start it up so it's running in the background. The status can be checked using `pg_isready`. If this fails, the installation can be started manually, e.g. via `pg_ctl -D /usr/local/var/postgres -l /usr/local/var/postgres/server.log start`, depending on the location of your installation. If PostgreSQL was installed from a package manager, it often spits out a line at the end about this (i.e. "run this magical incantation to start the daemon").

If you think this is an ok suggestion, and decide to put the `pg_isready` note above in the dependencies, it might also be appropriate to move the `ruby -v` check up there as well.

**Comment by ScottKolo** [comment]
With a little of my own wordsmithing, I have added information about starting PostgreSQL and installing Ruby to the README. Let me know if you think it's too much or not enough.

scottkolo/suitesparse-matrix-collection-website@384f0ad354c2d5be1b0351bb3aec5c9a0adc0955

**Comment by ScottKolo** [comment]
Just following up - is there any further feedback or any actions required on our part? What are the next steps?

**Comment by jedbrown** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01244/joss.01244/10.21105.joss.01244.pdf)

**Comment by jedbrown** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Attempting to check references...
```

**Comment by whedon** [comment]
```Reference check summary:

OK DOIs

- 10.1145/2049662.2049663 is OK

MISSING DOIs

- https://doi.org/10.1016/s0092-8240(05)80013-8 may be missing for title: Efficient, high-quality force-directed graph drawing
- https://doi.org/10.1016/j.jpdc.2015.06.005 may be missing for title: An improved exact algorithm and an NP-completeness proof for sparse matrix bipartitioning
- https://doi.org/10.1007/978-1-4757-3145-3_17 may be missing for title: Algorithm XXX: Mongoose, A Graph Coarsening and Partitioning Library
- https://doi.org/10.2172/1128467 may be missing for title: A Projection Method for Metric-Constrained Optimization

INVALID DOIs

- None
```

**Comment by jedbrown** [comment]
Whedon is quite flaky about checking DOIs; they look fine in the paper. I made a PR for some bib formatting issues: https://github.com/ScottKolo/suitesparse-matrix-collection-website/pull/6
I think we'll be ready to archive and accept if that looks reasonable. Thanks for your patience.

**Comment by arfon** [comment]
> Whedon is quite flaky about checking DOIs

I think Whedon only checks DOIs for `@article` entries currently. I'll see if I can fix this.

**Comment by arfon** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Attempting to check references...
```

**Comment by whedon** [comment]
```Reference check summary:

OK DOIs

- 10.1145/2049662.2049663 is OK
- 10.1007/978-1-5041-2940-4_9 is OK
- 10.17616/R3WG6Z is OK

MISSING DOIs

- https://doi.org/10.1145/1057588.1057590 may be missing for title: The Rutherford-Boeing sparse matrix collection
- https://doi.org/10.14744/anatoljcardiol.2017.7846 may be missing for title: Data Citation Index - Clarivate Analytics
- https://doi.org/10.1007/978-1-4842-2589-9_10 may be missing for title: University of Florida Sparse Matrix Collection - AWS Public Data Set
- https://doi.org/10.1016/s0092-8240(05)80013-8 may be missing for title: Efficient, high-quality force-directed graph drawing
- https://doi.org/10.1016/j.jpdc.2015.06.005 may be missing for title: An improved exact algorithm and an NP-completeness proof for sparse matrix bipartitioning
- https://doi.org/10.1007/978-1-4757-3145-3_17 may be missing for title: Algorithm XXX: Mongoose, A Graph Coarsening and Partitioning Library
- https://doi.org/10.2172/1128467 may be missing for title: A Projection Method for Metric-Constrained Optimization

INVALID DOIs

- None
```

**Comment by ScottKolo** [comment]
I have checked all of the suggested DOIs - they are all incorrect. Here's a quick discussion about each of these references and why they (currently) lack DOIs:

>MISSING DOIs
>
>- https://doi.org/10.1145/1057588.1057590 may be missing for title: The Rutherford-Boeing sparse matrix collection

This is a tech report. The suggested DOI references a paper that predates it by 15 years.

>- https://doi.org/10.14744/anatoljcardiol.2017.7846 may be missing for title: Data Citation Index - Clarivate Analytics

This is a website for Clarivate Analytics' Data Citation Index. The suggested DOI leads to a cardiology journal (?).

>- https://doi.org/10.1007/978-1-4842-2589-9_10 may be missing for title: University of Florida Sparse Matrix Collection - AWS Public Data Set

This is a website describing the use of the UFL Sparse Matrix Collection via AWS. The suggested DOI points to a generic article on Database Services in AWS.

>- https://doi.org/10.1016/s0092-8240(05)80013-8 may be missing for title: Efficient, high-quality force-directed graph drawing

This article was published in the Mathematica Journal, but it does not seem to have any DOI associated with it. The suggested DOI is a broken link.

>- https://doi.org/10.1016/j.jpdc.2015.06.005 may be missing for title: An improved exact algorithm and an NP-completeness proof for sparse matrix bipartitioning

This article is currently a preprint on arXiv. The suggested DOI links to some prior work by the same authors.

>- https://doi.org/10.1007/978-1-4757-3145-3_17 may be missing for title: Algorithm XXX: Mongoose, A Graph Coarsening and Partitioning Library

This article is currently in review. The suggested DOI references work by similar authors almost 20 years ago.

>- https://doi.org/10.2172/1128467 may be missing for title: A Projection Method for Metric-Constrained Optimization

This article is a preprint on arXiv. The suggested DOI points to an unrelated article in physics.

**Comment by ScottKolo** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01244/joss.01244/10.21105.joss.01244.pdf)

**Comment by jedbrown** [comment]
Yeah, the DOI checker is very buggy. References look good to me now. Can you fix one spelling error "constrainted", tag a release ([annotated tag](https://git-scm.com/book/en/v2/Git-Basics-Tagging#_annotated_tags) preferred), and archive on Zenodo or similar?

**Comment by ScottKolo** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01244/joss.01244/10.21105.joss.01244.pdf)

**Comment by ScottKolo** [comment]
One last nitpick: Is there any way to prevent line breaks in the middle of URLs?

![image](https://user-images.githubusercontent.com/3859035/54043531-75dd9f80-4192-11e9-8617-2b6c4ea26bab.png)

**Comment by jedbrown** [comment]
I would consider dropping the `https://` part because it's arguably clutter given that full links are embedded. and/or using a tt font:
```
[`https://sparse.tamu.edu/Mycielski`](https://sparse.tamu.edu/Mycielski)
```

**Comment by ScottKolo** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01244/joss.01244/10.21105.joss.01244.pdf)

**Comment by ScottKolo** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01244/joss.01244/10.21105.joss.01244.pdf)

**Comment by ScottKolo** [comment]
Okay, I have tagged release at JOSS-1.0.2. 
https://github.com/ScottKolo/suitesparse-matrix-collection-website/releases/tag/JOSS-1.0.2

We have archived the code for this release using figshare: https://figshare.com/articles/SuiteSparse_Matrix_Collection_Website/7381076

Does all of that look correct?

**Comment by jedbrown** [comment]
Author list on the figshare appears to be missing Yifan Hu, who was on the paper. Can you fix the author lists to match in content and order?

**Comment by ScottKolo** [comment]
Fixed. The Figshare DOI is https://doi.org/10.6084/m9.figshare.7381076

The title now matches, as do the authors and author order.

**Comment by jedbrown** [comment]
@whedon set JOSS-1.0.2 as version

**Comment by whedon** [comment]
OK. JOSS-1.0.2 is the version.

**Comment by jedbrown** [comment]
@ScottKolo Is it intended that `https://` appears in exactly one of the URLs?

**Comment by ScottKolo** [comment]
Yes. We removed it for most as you suggested, since it wasn’t helpful in those contexts, but left it when we referenced its deployment location.

**Comment by jedbrown** [comment]
@whedon set 10.6084/m9.figshare.7381076 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.6084/m9.figshare.7381076" target="_blank">10.6084/m9.figshare.7381076</a> is the archive.

**Comment by jedbrown** [comment]
Over to you, @openjournals/joss-eics.

**Comment by arfon** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/543

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/543, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by whedon** [comment]
```Reference check summary:

OK DOIs

- 10.1145/2049662.2049663 is OK
- 10.1007/978-1-5041-2940-4_9 is OK
- 10.17616/R3WG6Z is OK

MISSING DOIs

- https://doi.org/10.1145/1057588.1057590 may be missing for title: The Rutherford-Boeing sparse matrix collection
- https://doi.org/10.4275/kslis.2016.50.1.189 may be missing for title: Data Citation Index
- https://doi.org/10.1007/978-1-4842-2589-9_10 may be missing for title: University of Florida Sparse Matrix Collection - AWS Public Data Set
- https://doi.org/10.1016/s0092-8240(05)80013-8 may be missing for title: Efficient, high-quality force-directed graph drawing
- https://doi.org/10.1016/j.jpdc.2015.06.005 may be missing for title: An improved exact algorithm and an NP-completeness proof for sparse matrix bipartitioning
- https://doi.org/10.1007/978-1-4757-3145-3_17 may be missing for title: Algorithm XXX: Mongoose, A Graph Coarsening and Partitioning Library
- https://doi.org/10.2172/1128467 may be missing for title: A Projection Method for Metric-Constrained Optimization

INVALID DOIs

- None
```

**Comment by arfon** [comment]
@ScottKolo - the paper currently looks a little peculiar with both the figures being right at the start. This PR moves them into the body of the paper: https://github.com/ScottKolo/suitesparse-matrix-collection-website/pull/7

**Comment by arfon** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```Reference check summary:

OK DOIs

- 10.1145/2049662.2049663 is OK
- 10.1007/978-1-5041-2940-4_9 is OK
- 10.17616/R3WG6Z is OK

MISSING DOIs

- https://doi.org/10.1145/1057588.1057590 may be missing for title: The Rutherford-Boeing sparse matrix collection
- https://doi.org/10.4275/kslis.2016.50.1.189 may be missing for title: Data Citation Index
- https://doi.org/10.1007/978-1-4842-2589-9_10 may be missing for title: University of Florida Sparse Matrix Collection - AWS Public Data Set
- https://doi.org/10.1016/s0092-8240(05)80013-8 may be missing for title: Efficient, high-quality force-directed graph drawing
- https://doi.org/10.1016/j.jpdc.2015.06.005 may be missing for title: An improved exact algorithm and an NP-completeness proof for sparse matrix bipartitioning
- https://doi.org/10.1007/978-1-4757-3145-3_17 may be missing for title: Algorithm XXX: Mongoose, A Graph Coarsening and Partitioning Library
- https://doi.org/10.2172/1128467 may be missing for title: A Projection Method for Metric-Constrained Optimization

INVALID DOIs

- None
```

**Comment by whedon** [comment]
Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/544

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/544, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by arfon** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/545
1. Wait a couple of minutes to verify that the paper DOI resolves [https://doi.org/10.21105/joss.01244](https://doi.org/10.21105/joss.01244)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? notify your editorial technical team...

**Comment by arfon** [comment]
@dblarremore, @tkonolige - many thanks for your reviews and to @jedbrown for editing this submission ✨ 

@ScottKolo - your paper is now accepted into JOSS :zap::rocket::boom:

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](http://joss.theoj.org/papers/10.21105/joss.01244/status.svg)](https://doi.org/10.21105/joss.01244)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.01244">
  <img src="http://joss.theoj.org/papers/10.21105/joss.01244/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: http://joss.theoj.org/papers/10.21105/joss.01244/status.svg
   :target: https://doi.org/10.21105/joss.01244
```
This is how it will look in your documentation:

[![DOI](http://joss.theoj.org/papers/10.21105/joss.01244/status.svg)](https://doi.org/10.21105/joss.01244)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: http://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://www.flipcause.com/secure/cause_pdetails/Mjk3Nzk=
